### PR TITLE
Add free form map config for task custom

### PR DESCRIPTION
**Pull request by katrogan**
Signed-off-by: Katrina Rogan <katroganGH@gmail.com>

# TL;DR
Add free form config for task custom

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/816

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by wild-endeavor** [comment]
Without really understanding what the pod task change is, would this change be relevant if we passed down the entire task template? @kumare3

**Comment by kumare3** [comment]
> Without really understanding what the pod task change is, would this change be relevant if we passed down the entire task template? @kumare3 

Nope, this is for the backend- and greatly simplifying custom

**Comment by katrogan** [comment]
ping @kumare3 @EngHabu

File: protos/flyteidl/core/tasks.proto [hunk 1]
```diff
@@ -125,6 +125,12 @@ message TaskTemplate {
 
     // security_context encapsulates security attributes requested to run this task.
     SecurityContext security_context = 8;
+
+    // Metadata about the custom defined for this task. This is extensible to allow various plugins in the system
+    // to use as required.
+    // reserve the field numbers 1 through 15 for very frequently occurring message elements
+    map<string, string> custom_config = 16;
```
Comment (kumare3) [inline protos/flyteidl/core/tasks.proto:132]:
```suggestion
    map<string, string> config = 16;
```

Comment (katrogan) [reply protos/flyteidl/core/tasks.proto]: done

**Comment by katrogan** [comment]
@kumare3 updated PTAL

File: protos/flyteidl/core/tasks.proto [hunk 2]
```diff
@@ -125,6 +125,12 @@ message TaskTemplate {
 
     // security_context encapsulates security attributes requested to run this task.
     SecurityContext security_context = 8;
+
+    // Metadata about the custom defined for this task. This is extensible to allow various plugins in the system
+    // to use as required.
+    // reserve the field numbers 1 through 15 for very frequently occurring message elements
+    map<string, string> config = 16;
```
Comment (EngHabu) [inline protos/flyteidl/core/tasks.proto:132]: Should we mark the config inside the container as deprecated?

Comment (katrogan) [reply protos/flyteidl/core/tasks.proto:132]: sure, done @EngHabu

File: protos/flyteidl/core/tasks.proto [hunk 3]
```diff
@@ -156,7 +162,7 @@ message Container {
 
     // Allows extra configs to be available for the container.
     // TODO: elaborate on how configs will become available.
```
Comment (EngHabu) [inline protos/flyteidl/core/tasks.proto:164]:
```suggestion
    // TODO: elaborate on how configs will become available.
    // Deprecated, please use TaskTemplate.config instead.
```

**Comment by katrogan** [comment]
ping @EngHabu

**Review by EngHabu — APPROVED** [review]

**Pull request merged**